src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -427,6 +508,24 @@ public void checkResponse (String header, String method, URL url)
         checkResponse (header, method, url.getFile());
     }
 
+    private void validateAlgorithm(String algorithm) throws IOException {
+        if (getAuthType() == AuthCacheValue.Type.Server &&
+                getProtocolScheme().equals("https")) {
+            // HTTPS server authentication can use any algorithm
```

Comment:
A more security conscious user may want to disable MD5 digest authentication, even when used over HTTPS, even though the risks are far less than when used over HTTP. Is there a way to do that?

Reply:
There isn't a way to do that currently. I'd prefer not to further complicate the configuration to allow it. It would probably be a lot simpler to just disable MD5 across the board for proxy and server instead? We're including SHA-1 in that list now as well by the way.